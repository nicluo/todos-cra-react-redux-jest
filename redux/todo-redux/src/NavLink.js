import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NavLink extends Component {
  render() {
    const {to, children} = this.props;
    return (
      <a className="link dim white dib mh3" href={to} title={children}>
        {children}
      </a>
    )
  }
}

NavLink.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired
};

export default NavLink;
