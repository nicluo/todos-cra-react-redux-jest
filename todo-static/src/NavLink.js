import React, { Component } from 'react';

class NavLink extends Component {
  render() {
    return (
      <a className="link dim white dib" href="#" title="Todo">Todo</a>
    )
  }
}

export default NavLink;
