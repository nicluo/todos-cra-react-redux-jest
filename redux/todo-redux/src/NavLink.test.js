import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import NavLink from './NavLink';

it('Renders brand link correctly', () => {
  const component = shallow(
    <NavLink to="/">Todo</NavLink>
  );
  expect(shallowToJson(component)).toMatchSnapshot();
});

it('Renders another link correctly', () => {
  const component = shallow(
    <NavLink to="/archive">Archive</NavLink>
  );
  expect(shallowToJson(component)).toMatchSnapshot();
});
