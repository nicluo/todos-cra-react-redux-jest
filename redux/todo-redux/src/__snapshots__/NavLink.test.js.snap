// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders another link correctly 1`] = `
<a
  className="link dim white dib mh3"
  href="/archive"
  title="Archive"
>
  Archive
</a>
`;

exports[`Renders brand link correctly 1`] = `
<a
  className="link dim white dib mh3"
  href="/"
  title="Todo"
>
  Todo
</a>
`;
